src/cpp/lpnamer/problem_modifier/AntaresProblemToXpansionProblemTranslator.cpp

```diff
@@ -5,11 +5,109 @@
 #include "antares-xpansion/lpnamer/problem_modifier/AntaresProblemToXpansionProblemTranslator.h"
 
 #include <algorithm>
+#include <charconv>
 #include <cmath>
+#include <regex>
 
 #include "antares-xpansion/multisolver_interface/SolverFactory.h"
 #include "antares-xpansion/xpansion_interfaces/LogUtils.h"
 
+constexpr unsigned int HOURS_IN_A_WEEK = 168;
+constexpr unsigned int DAYS_IN_A_WEEK = 7;
+
+static bool try_replace_first_token(const std::string& name,
+                                    std::string_view prefix,
+                                    unsigned int week,
+                                    long long factor,
+                                    bool ignore_value,
+                                    std::string& out)
+{
+    const auto pos = name.find(prefix);
+    if (pos == std::string::npos)
+    {
+        return false;
+    }
+
+    const auto start = pos + prefix.size();
+    const auto end = name.find('>', start);
+    if (end == std::string::npos)
+    {
+        throw std::runtime_error(LOGLOCATION + std::string("Malformed token: missing '>'"));
+    }
+
+    long long original = 0;
+    if (!ignore_value)
+    {
+        const auto digits = name.substr(start, end - start);
+        auto first = digits.data();
+        auto last = digits.data() + digits.size();
+        auto ec = std::from_chars(first, last, original);
+        if (ec.ec != std::errc() || ec.ptr != last)
+        {
+            throw std::runtime_error(LOGLOCATION
+                                     + std::string("Malformed token: non-integer value"));
+        }
+    }
+
+    const long long new_value = ignore_value
+                                  ? static_cast<long long>(week) - 1
+                                  : (static_cast<long long>(week) - 1) * factor + original;
+
+    out = name;
+    out.replace(pos, (end - pos) + 1, std::string(prefix) + std::to_string(new_value) + ">");
+    return true;
+}
+
+static std::string replace_hour_in_name(const std::string& name, unsigned int week)
+{
+    if (week == 0)
+    {
+        throw std::invalid_argument(LOGLOCATION + std::string("week must be >= 1"));
+    }
+
+    std::string out;
+    if (try_replace_first_token(name, "hour<", week, HOURS_IN_A_WEEK, false, out))
+    {
+        return out;
+    }
+    if (try_replace_first_token(name, "day<", week, DAYS_IN_A_WEEK, false, out))
+    {
+        return out;
+    }
+    if (try_replace_first_token(name, "week<", week, 1, true, out))
+    {
+        return out;
+    }
+
+    throw std::runtime_error(LOGLOCATION + std::string("No [hour|day|week]<...> pattern found in ")
+                             + name);
+}
+
+/**
+ * Static variables to store renamed variables and constraints names
+ * If we had an object it would be private members
+ */
+std::unordered_map<int, std::vector<std::string>> variables_names;
+std::unordered_map<int, std::vector<std::string>> constraints_names;
```

Comment:
For 1 year, you don't seem to need this. `variables_names[week]` seems to be used only once.
Can you confirm that these variables are re-used for years 2 and onward ?
Also, global variables bad